web/src/compose_recipient.ts

```diff
@@ -37,24 +36,14 @@ function composing_to_current_topic_narrow(): boolean {
 }
 
 function composing_to_current_private_message_narrow(): boolean {
-    const compose_state_recipient = compose_state.private_message_recipient_emails();
-    const narrow_state_recipient = narrow_state.pm_emails_string();
-    if (narrow_state_recipient === undefined) {
+    const compose_state_recipient = new Set(compose_state.private_message_recipient_ids());
+    const narrow_state_recipient = narrow_state.pm_ids_set();
+    if (narrow_state_recipient.size === 0) {
         return false;
     }
     return (
-        Boolean(compose_state_recipient) &&
-        Boolean(narrow_state_recipient) &&
-        _.isEqual(
-            compose_state_recipient
-                .split(",")
-                .map((s) => s.trim())
-                .sort(),
-            narrow_state_recipient
-                .split(",")
-                .map((s) => s.trim())
-                .sort(),
-        )
+        narrow_state_recipient.intersection(compose_state_recipient).size ===
+        narrow_state_recipient.size
```

Comment:
This only checks that `narrow_state_recipient` is a subset of `compose_state_recipient`. That is NOT an equality check.

Reply:
oh of course. cc @amanagr who [suggested this](https://github.com/zulip/zulip/pull/34628#discussion_r2090915219) (unless I misunderstood). 

I think it would be cleaner to change it back to comparing sorted arrays

Reply:
It's worth noting these lists are usually shorter than 10 users

Reply:
![image](https://github.com/user-attachments/assets/16bba5b8-d23e-4432-b6b8-59ff1b13ccb4)

`_.isEqual` works for set comparison.

Reply:
Great, let's do that

Reply:
I also updated a few other instance of ` _.isEqual(util.sorted_ids ...` in this PR